extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -761,6 +767,69 @@ export class SchemaCompareResult {
 		});
 	}
 
+	private createSaveScmpButton(view: azdata.ModelView): void {
+		this.saveScmpButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.saveScmpButton', 'Save .scmp file'),
+			iconPath: {
+				light: path.join(__dirname, 'media', 'save-scmp.svg'),
+				dark: path.join(__dirname, 'media', 'save-scmp-inverse.svg')
+			},
+			title: localize('schemaCompare.saveScmpButtonTitle', 'Save source and target, options, and excluded elements'),
+			enabled: false
+		}).component();
+
+		this.saveScmpButton.onDidClick(async (click) => {
+			const rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
+			const filePath = await vscode.window.showSaveDialog(
+				{
+					defaultUri: vscode.Uri.file(rootPath),
+					saveLabel: localize('schemaCompare.saveFile', 'Save'),
+					filters: {
+						'scmp Files': ['scmp'],
+					}
+				}
+			);
+
+			if (!filePath) {
+				return;
+			}
+
+			// convert include/exclude maps to arrays of object ids
+			let sourceExcludes: azdata.SchemaCompareObjectId[] = this.convertExcludesToObjectIds(this.originalSourceExcludes);
+			let targetExcludes: azdata.SchemaCompareObjectId[] = this.convertExcludesToObjectIds(this.originalTargetExcludes);
+
+			Telemetry.sendTelemetryEvent('SchemaCompareSaveScmp');
+			const service = await SchemaCompareResult.getService(msSqlProvider);
+			const result = await service.schemaCompareSaveScmp(this.sourceEndpointInfo, this.targetEndpointInfo, azdata.TaskExecutionMode.execute, this.deploymentOptions, filePath.fsPath, sourceExcludes, targetExcludes);
+			if (!result || !result.success) {
+				Telemetry.sendTelemetryEvent('SchemaCompareSaveScmpFailed', {
+					'errorType': getTelemetryErrorType(result.errorMessage),
+					'operationId': this.comparisonResult.operationId
+				});
+				vscode.window.showErrorMessage(
+					localize('schemaCompare.saveScmpErrorMessage', "Save scmp failed: '{0}'", (result && result.errorMessage) ? result.errorMessage : 'Unknown'));
+			}
+
+			Telemetry.sendTelemetryEvent('SchemaCompareSaveScmpEnded', {
+				'endTime:': Date.now().toString(),
+				'operationId': this.comparisonResult.operationId
+			});
+		});
+	}
+
+	// converts excluded diff entries into object ids which are needed to save them in an scmp
+	private convertExcludesToObjectIds(exludes: Map<string, azdata.DiffEntry>): azdata.SchemaCompareObjectId[] {
+		let result = [];
+		exludes.forEach((value: azdata.DiffEntry) => {
+			result.push({
+				nameParts: value.sourceValue ? value.sourceValue : value.targetValue,
+				sqlObjectType: `Microsoft.Data.Tools.Schema.Sql.SchemaModel.${value.name}`
```

Comment:
Good workaround but little scary. 
I wish in hindsight that we should have added this info to diffEntry as well. Will add a later task for that.